src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,238 @@
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars SET model = ?, "
+                + "manufacturer_id = ? "
+                + "WHERE id = ? AND deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                  PreparedStatement updateStatement = connection.prepareStatement(query)) {
+            updateStatement.setString(1, car.getModel());
+            updateStatement.setLong(2, car.getManufacturer().getId());
+            updateStatement.setLong(3, car.getId());
+            if (updateStatement.executeUpdate() < 1) {
+                throw new NoSuchElementException("Can`t find element by car" + car);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t update car " + car, e);
+        }
+        deleteCarsDriversDependency(car.getId());
+        for (Driver driver : car.getDrivers()) {
+            setCarDrivers(car.getId(), driver.getId());
+        }
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET deleted = true "
+                + "WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteStatement = connection.prepareStatement(query)) {
+            deleteStatement.setLong(1, id);
+            return deleteStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t delete Car by id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT c.id, c.manufacturer_id, c.model, cd.driver_id FROM cars c "
+                + "LEFT JOIN cars_drivers cd "
+                + "ON c.id = cd.car_id "
+                + "WHERE cd.driver_id = ?;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllByDriverStatement = connection.prepareStatement(query)) {
+            getAllByDriverStatement.setLong(1, driverId);
+            ResultSet resultSet = getAllByDriverStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(createCar(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t get List of all drivers by id " + driverId, e);
+        }
+        for (Car car: cars) {
+            car.setManufacturer(getManufacture(car.getId()));
+            car.setDrivers(getAllDriversList(car.getId()));
+        }
+        return cars;
+    }
+
+    private Car createCar(ResultSet resultSet) throws SQLException {
+        Car car = new Car();
+        car.setId(resultSet.getObject("id", Long.class));
+        car.setModel(resultSet.getString("model"));
+        return car;
+    }
+
+    private Manufacturer getManufacture(Long id) {
```

Comment:
you don't need to make any queries here, you do one query in the get() method and you get ResultSet you it instead
but I would recommend moving it to createCar() method